pytorch_lightning/utilities/parsing.py

```diff
@@ -197,7 +198,11 @@ def save_hyperparameters(
 
     if not frame:
         frame = inspect.currentframe().f_back
-    init_args = get_init_args(frame)
+
+    if is_dataclass(obj):
```

Comment:
@awaelchli I don't think we are using this `save_hyperparameters` function, but the one in the Lightning module. I guess this was decoupled for an use case in Flash.

Reply:
in what sense do you mean we are not using it. the test that I added fails on master.

Reply:
Nvm, just checked, it was divided.